deploy/helm/metrics-discovery-config.yaml

```diff
@@ -0,0 +1,268 @@
+    directory: /var/lib/swo/sending_queue
+{{- end }}
+  health_check:
+    endpoint: 0.0.0.0:13133
+  k8s_observer:
+    auth_type: serviceAccount
+    observe_pods: true    
+
+processors:
+  swk8sattributes:
+{{ include "common.k8s-instrumentation" . | indent 4 }}
+
+  memory_limiter:
+{{ toYaml .Values.aws_fargate.metrics.autodiscovery.memory_limiter | indent 4 }}
+
+  transform/scope:
+    metric_statements:
+      - context: scope
+        statements:
+          - set(name, "")
+          - set(version, "")
+
+  metricstransform/rename/discovery:
+    transforms:
+      # add `k8s.` prefix to all metrics
+      - include: ^(.*)$$
+        match_type: regexp
+        action: update
+        new_name: {{ .Values.otel.metrics.autodiscovery.prefix }}$${1}
+
+ 
+  groupbyattrs/common-all:
+    keys:
+      - k8s.container.name
+      - k8s.namespace.name
+      - k8s.pod.name
+      - k8s.pod.uid
+      - k8s.node.name
+      - host.name
+      - service.name
+  
+  resource/all:
+    attributes:
+      # Collector and Manifest version
+      - key: sw.k8s.agent.manifest.version
+        value: ${MANIFEST_VERSION}
+        action: insert
+
+      - key: sw.k8s.agent.app.version
+        value: ${APP_VERSION}
+        action: insert
+
+      # Cluster
+      - key: sw.k8s.cluster.uid
+        value: ${CLUSTER_UID}
+        action: insert
+
+      - key: k8s.cluster.name
+        value: ${CLUSTER_NAME}
+        action: insert
+ 
+
+{{- if .Values.otel.metrics.filter }}
+  filter/metrics:
+    metrics:
+{{ toYaml .Values.otel.metrics.filter | indent 6 }}
+{{- end }}
+
+  batch/metrics:
+{{ toYaml .Values.aws_fargate.metrics.autodiscovery.batch | indent 4 }}
+  
+{{- include "common-config.filter-remove-temporary-metrics" . | nindent 2 }}
+
+  filter/histograms:
+    metrics:
+      metric:
+        - 'type == METRIC_DATA_TYPE_HISTOGRAM and not(name == "k8s.rest_client_request_duration_seconds" or name == "k8s.workqueue_queue_duration_seconds")'
+
+{{- if .Values.otel.metrics.autodiscovery.prometheusEndpoints.filter }}
+  filter/metrics-discovery:
+    metrics:
+{{ toYaml .Values.otel.metrics.autodiscovery.prometheusEndpoints.filter | indent 6 }}
+{{- end }}
+
+{{- if .Values.otel.metrics.autodiscovery.prometheusEndpoints.customTransformations.counterToRate }}
+  cumulativetodelta/discovery:
+    include:
+      metrics:
+{{- range .Values.otel.metrics.autodiscovery.prometheusEndpoints.customTransformations.counterToRate }}
+        - {{ . }}
+{{- end }}
+      match_type: strict
+  deltatorate/discovery:
+    metrics:
+{{- range .Values.otel.metrics.autodiscovery.prometheusEndpoints.customTransformations.counterToRate }}
+      - {{ . }}
+{{- end }}
+{{- end }}
+
+{{- if ne .Values.otel.metrics.autodiscovery.prefix "k8s." }}
+  # in case the prefix differs from "k8s." we need to copy the required metrics
+  # so that SWO built-in dashboards works correctly
+{{- $arrayOfRequiredMetrics := list 
+  "etcd_disk_backend_commit_duration_seconds"
```

Comment:
Are the etcd metrics relevant here?

Reply:
It would be absurd environment if someone with Fargate installs etcd separately, since etcd is managed by AWS. But for sake of consistency I would keep same metric list here